src/app/forms/custom-sections/weights/weights.component.ts

```diff
@@ -61,11 +63,50 @@ export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
     return FormNodeEditTypes;
   }
 
-  get axles(): FormArray {
-    return this.form.get(['axles']) as FormArray;
+  get axles(): CustomFormArray {
+    return this.form.get(['axles']) as CustomFormArray;
   }
 
-  getAxleWeights(i: number): FormGroup {
-    return this.axles.get([i, 'weights']) as FormGroup;
+  getAxleWeights(i: number): CustomFormGroup {
+    return this.axles.get([i, 'weights']) as CustomFormGroup;
+  }
+
+  addAxle(): void {
+    const newAxle: Axle = {
+      axleNumber: this.axles.length + 1,
+      tyres: {
+        tyreSize: null,
+        speedCategorySymbol: null,
+        fitmentCode: null,
+        dataTrAxles: null,
+        plyRating: null,
+        tyreCode: null
+      },
+      parkingBrakeMrk: false,
+      weights: {
+        kerbWeight: null,
+        ladenWeight: null,
+        gbWeight: null,
+        eecWeight: null,
+        designWeight: null
+      }
+    };
+    if (this.vehicleTechRecord.axles.length < 5) {
```

Comment:
nit: alot of the code here is validating the form or rather the model for the form, could we move this into a validator (not sure myself) and then leave these as dumb functions

Reply:
Would need to be a group validator which is pretty spicy

Reply:
Validators can be applied to form groups and form arrays as well .e.g. `new FormArray([], [Validators.required])` would enforce a positive value. Alternatively, create a custom validator for minLengthArray

Reply:
I agree there needs to be a tidy here, but I'm not sure adding a validator is the best plan? Really you don't want a user to delete an axle that they then cannot submit without. The validator will say like 'hey you need 2 axles' but it won't stop you actually removing it right? I think this is a bad UX in this case, because if they clicked it, they would need to re add which could be an intensive job. Any thoughts here?

Reply:
Happy to pause conversation for now but would be cool to talk about different validators that stop actions because up to now I agree with you nathan than we display some sort of error but the action can still be carried out